competitionsApp.py

```diff
@@ -385,6 +388,21 @@ def competition_admin_post(competition_id):
         competition['climbers'].pop(remove_climber)
         competitionsEngine.update_competition(competition['id'], competition)
 
+    if climber_id is not None:
+        competition['climbers'][climber_id]['name'] = request.form.get('name_'+ climber_id)
+        competition['climbers'][climber_id]['sex'] = request.form.get('sex_'+ climber_id)
+        competition['climbers'][climber_id]['club'] = request.form.get('club_'+ climber_id)
+        competition['climbers'][climber_id]['email'] = request.form.get('email_'+ climber_id)
+        try:
+            # Attempt to retrieve the category from the form data and convert it to an integer
+            competition['climbers'][climber_id]['category'] = int(request.form.get('category_' + str(climber_id)))
+        except (ValueError, TypeError):
+            # Handle the exception if the conversion fails
+            print("The provided category is not a valid integer for climber " + competition['climbers'][climber_id]['name'])
+            # Set to a default value or handle the error as appropriate
+            competition['climbers'][climber_id]['category'] = 0  # Replace default_value with whatever default you wish to use
+        competitionsEngine.update_competition(competition['id'], competition)
+
```

Comment:
We also need to update the main user table to persist these changes. The assumption is that the person registered with wrong club or wrong sex so we need to fix it but that same fix should be persisted to the main user table so that it's correct for next registration. 

So we need to do here what we do in competitionsApp.update_user:

climber = competitionsEngine.get_user_by_email(session.get('email'))
climber = competitionsEngine.user_self_update(climber, name, firstname, lastname, nick, sex, club, category)